webpack.config.common.js

```diff
@@ -0,0 +1,158 @@
+const path = require('path');
+const glob = require('glob');
+const MiniCssExtractPlugin = require("mini-css-extract-plugin");
+const RemoveEmptyScriptsPlugin = require('webpack-remove-empty-scripts');
+const PostCSSAssetsPlugin = require('postcss-assets-webpack-plugin');
+
+const rootDir = path.resolve(__dirname);
+const buildDir = 'dist/code-assets';
+
+// Define all entries:
+const globals = ['./css/global.scss'];
+const components = glob.sync('./templates/**/*.{scss,js}', {
+  dotRelative: true,
+  ignore: [
+    "./templates/**/_*",    // Exclude from the bundle process files starting with '_'.
+    "./templates/**/_*/**"  // Exclude from the bundle process entire directories starting with '_'.
+  ]}
+);
+
+// Declare an entry object that will be assigned as value to the webpack 'entry' key:
+const entry = {};
+
+const allMatches = [...globals, ...components];
+
+const localOutputPath = (filePath) => {
+  const fileDir = path.dirname(filePath);
+  const fileBasename = path.basename(filePath);
+  const fileExtension = path.extname(fileBasename);
+  const fileName = path.basename(fileBasename, fileExtension);
+  let fileExtensionDir = 'ERROR/';
+
+  switch (fileExtension) {
+    case '.js':
+      fileExtensionDir = 'js/';
+      break;
+
+    case '.scss':
+      fileExtensionDir = 'css/';
+      break;
+  }
+
+  return fileName + '/' + fileExtensionDir + fileName + '.processed';
+};
+
+allMatches.forEach((match) => {
+  let entryKey = localOutputPath(match);
+  entry[entryKey] = match;
+});
```

Comment:
Instead of defining an entry and assigning from forEach, it will be more declarative to use `reduce` on the `allMatches` and compose an object

Reply:
Fixed